src/Silk.Data/MediatR/Handlers/TagRequestHandler.cs

```diff
@@ -0,0 +1,166 @@
+            public async Task<Tag> Handle(TagRequest.Update request, CancellationToken cancellationToken)
+            {
+                Tag tag = await _db.Tags.FirstAsync(t => t.Name.ToLower() == request.Name.ToLower() &&
+                                                         t.GuildId == request.GuildId, cancellationToken);
+                tag.Name = request.NewName ?? tag.Name;
+                tag.Uses = request.Uses ?? tag.Uses;
+                tag.Content = request.Content ?? tag.Content;
+                tag.Aliases = request.Aliases ?? tag.Aliases;
+
+                await _db.SaveChangesAsync(cancellationToken);
+                return tag;
+            }
+        }
+
+        public class DeleteHandler : IRequestHandler<TagRequest.Delete>
+        {
+            private readonly SilkDbContext _db;
+            public DeleteHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+            
+            public async Task<Unit> Handle(TagRequest.Delete request, CancellationToken cancellationToken)
+            {
+                Tag tag = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.Aliases)
+                    .AsSplitQuery()
+                    .FirstOrDefaultAsync(t =>
+                        t.Name.ToLower() == request.Name.ToLower() &&
+                        t.GuildId == request.GuildId, cancellationToken);
+                
+                Tag[] aliasedTags = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.Aliases)
+                    .AsSplitQuery()
+                    .Where(t => t.OriginalTagId == tag.Id || t.OriginalTag!.OriginalTagId == tag.Id)
+                    .ToArrayAsync(cancellationToken);
+                
+                
+                foreach (Tag t in aliasedTags)
+                    _db.Tags.Remove(t);
+                
+                _db.Tags.Remove(tag);
+                await _db.SaveChangesAsync(cancellationToken);
+                return new();
+            }
+        }
+        
+        public class GetByUserHandler : IRequestHandler<TagRequest.GetByUser, IEnumerable<Tag>?>
+        {
+            private readonly SilkDbContext _db;
+            public GetByUserHandler(SilkDbContext db)
+            {
+                _db = db;
+            }
+            
+            public async Task<IEnumerable<Tag>?> Handle(TagRequest.GetByUser request, CancellationToken cancellationToken)
+            { 
+                Tag[] tags = await _db
+                    .Tags
+                    .Include(t => t.OriginalTag)
+                    .Include(t => t.OriginalTag)
+                    .Where(t => t.GuildId == request.GuildId && t.OwnerId == request.OwnerId)
+                    .ToArrayAsync(cancellationToken);
+                
+                return tags.Any() ? tags : null; // Return null over empty list //
```

Comment:
null over an empty list seems like an odd convention?  More explosive.  But it's not wrong.

Reply:
Saves from having to iterate over a collection more than necessary. I also like the `is null` syntax.